controller_interface/src/controller_interface_base.cpp

```diff
@@ -92,22 +93,13 @@ return_type ControllerInterfaceBase::init(
     return return_type::ERROR;
   }
 
-  switch (on_init())
-  {
-    case LifecycleNodeInterface::CallbackReturn::SUCCESS:
-      break;
-    case LifecycleNodeInterface::CallbackReturn::ERROR:
-    case LifecycleNodeInterface::CallbackReturn::FAILURE:
-      RCLCPP_DEBUG(
-        get_node()->get_logger(),
-        "Calling shutdown transition of controller node '%s' due to init failure.",
-        get_node()->get_name());
-      node_->shutdown();
-      return return_type::ERROR;
-  }
-
   node_->register_on_configure(
-    std::bind(&ControllerInterfaceBase::on_configure, this, std::placeholders::_1));
+    [this](const rclcpp_lifecycle::State & previous_state) -> CallbackReturn
+    {
+      lifecycle_id_.store(this->get_lifecycle_state().id(), std::memory_order_release);
+      const auto return_value = on_configure(previous_state);
+      return return_value;
```

Comment:
Why this pattern and not directly return
```suggestion
      return on_configure(previous_state);
```
or shouldn't the lifecycle_id_ be stored afterwards instead?